### DISCUSSION TITLE
How to integtrate full Peppol message validation

### DISCUSSION THREAD

**Discussion by kukel** [OxalisCommunity/oxalis#541] [Ideas]
I have been looking into full Peppol message validation before sending out messages or when receiving. I (think I) want this to be integrated into Oxalis and not as a standalone post- or pre- service. I am unable to find the right place in the code for this. Is there a good way/hook to do this or should I take a different approach?

Part of this is related to https://docs.peppol.eu/poacc/upgrade-3/profiles/36-mlr/

Any thoughts?

**Comment by aaron-kumar** [comment]
Oxalis is [Access Point software implementation](https://peppol.eu/downloads/peppolimplementations/) (AS2 and AS4) based on [Access Point Specification](https://peppol.eu/downloads/the-peppol-edelivery-network-specifications/)
Adding message validation support for global/country specific rules is outside the scope of Oxalis (_Maintaining those validation require extra effort, so better to keep the solution/library limited to what it is meant for_) . 
But if you still want to add it as an add-on component then use Oxalis Plugin architecture to extend this. Feel free to contribute and share it with rest of community.  

P.S. I am converting this issue to discussion as it is not bug.

**Reply by kukel** [reply]
I know that Oxalis is an AccessPoint and I know it is a 'shall' and not a 'must' but 4.2.3 of the compliance policy states 

> When acting as a sending Access Point, the PEPPOL AP Provider shall however monitor the 
behaviour and performance of the sender to ensure that business documents are confirmed as 
valid instances according to the applicable rules and technical specifications. 

That is what I'd like to cater for and yes, it can be done as a wrapper/separate step and maybe via the custom oxalis plugin mechanism.  Not sure if I'd like to go that direction. 

Thanks anyway.
